oauth2_provider/management/commands/create_oauth2_client.py

```diff
@@ -0,0 +1,159 @@
+import json
+from optparse import make_option
+
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.core.management.base import BaseCommand, CommandError
+from django.core.validators import URLValidator
+from provider.constants import CONFIDENTIAL, PUBLIC
+from provider.oauth2.models import Client
+
+from oauth2_provider.models import TrustedClient
+
+
+ARG_STRING = '<url> <redirect_uri> <client_type: "confidential" | "public">'
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    help = 'Create a new OAuth2 Client. Outputs a serialized representation of the newly-created Client.'
+    args = ARG_STRING
+    fields = None
+
+    option_list = BaseCommand.option_list + (
+        make_option(
+            '-u',
+            '--username',
+            action='store',
+            type='string',
+            dest='username',
+            help="Username of a user to associate with the Client."
+        ),
+        make_option(
+            '-n',
+            '--client_name',
+            action='store',
+            type='string',
+            dest='client_name',
+            help="String to assign as the Client name."
+        ),
+        make_option(
+            '-i',
+            '--client_id',
+            action='store',
+            type='string',
+            dest='client_id',
+            help="String to assign as the Client ID."
+        ),
+        make_option(
+            '-s',
+            '--client_secret',
+            action='store',
+            type='string',
+            dest='client_secret',
+            help="String to assign as the Client secret. Should not be shared."
+        ),
+        make_option(
+            '-t',
+            '--trusted',
+            action='store_true',
+            dest='trusted',
+            default=False,
+            help="Designate the Client as trusted. Trusted Clients bypass the user consent "
+                 "form typically displayed after validating the user's credentials."
+        ),
+    )
+
+    def handle(self, *args, **options):
+        self._clean_args(args)
+        self._parse_options(options)
+
+        trusted = self.fields.pop('trusted')
+
+        try:
+            client = Client.objects.create(**self.fields)
+        except Exception as e:
+            raise CommandError("Failed to create Client: {}".format(e))
```

Comment:
since the exception could have been anything, and `str(e)` might not be as informative as the actual stack trace, i'd suggest letting the original exception fly free here.